SuperBuild/External_LZMA.cmake

```diff
@@ -8,27 +8,32 @@ set(${proj}_DEPENDENCIES "")
 ExternalProject_Include_Dependencies(${proj} PROJECT_VAR proj DEPENDS_VAR ${proj}_DEPENDENCIES)
 
 if(Slicer_USE_SYSTEM_${proj})
-  unset(${proj}_DIR CACHE)
-  unset(${proj} CACHE)
-  find_package(${proj} REQUIRED)
+  unset(LZMA_INCLUDE_DIR CACHE)
```

Comment:
Is there a reason for using LZMA instead of ${proj}?

Reply:
* This allow easy grep for these variables in the code base 
* Consistent with the other external projects

Reply:
It is important to list all the variables we set for grepping, but that could be done once, in the documentation of the script. Right now, ${proj} and project name seems to be randomly used throughout all External_*.cmake files (I could not deduce any logic of why one or the other is used).

I prefer using `${proj}` as much as possible, because it removes the need for copy/paste and search/replace when creating or updating External_*.cmake files and it makes them easier to compare (you only see the real differences). For example, try to compare External_sqlite.cmake (it uses ${proj} everywhere) and External_lzma.cmake files with/without using ${proj} everywhere.

Reply:
Sounds good, I will update then. Also, it should be `External_LZMA.cmake`  instead of `External_lzma.cmake`

Reply:
Thanks you, sounds good!

Just curious, how do you decide between lzma and LZMA?

Reply:
I used LZMA to be consistent with the the variable names  (e.g `LZMA_INCLUDE_DIR`)

Usually, it is also good to be consistent with the name of the CMake project (which would be `lzma`. See [here](https://github.com/nextgis-borsch/lib_lzma/blob/02e9a98e1771391f4f98df6153ff9b0c8188a52d/CMakeLists.txt#L29)).

But the official  project reference it as `LZMA` (see https://sourceforge.net/p/lzmautils/wiki/Home/) and there are occurrences of FindLZMA.cmake (for example [here](https://github.com/microsoft/vcpkg/blob/master/ports/vtk/FindLZMA.patch))